skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceListener1.scala

```diff
@@ -0,0 +1,77 @@
+package com.evolutiongaming.skafka.consumer
+
+import cats.data.NonEmptySet
+import cats.~>
+import com.evolutiongaming.skafka.TopicPartition
+
+/**
+  * Will be converted to [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]] during consumer.subscribe
+  *
+  * Uses [[RebalanceCallback]] to describe the actions to be performed during rebalance.
+  *
+  * Please refer to [[RebalanceCallback]] for more details.
+  *
+  * Below is an example inspired by [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]] documentation.
+  *
+  * Compiling and working example is available [[https://github.com/evolution-gaming/skafka/blob/d2af038b012523533f2b73d432721d6d1e7cebbe/skafka/src/test/scala/com/evolutiongaming/skafka/consumer/RebalanceListener1Spec.scala here]]:
+  * TODO: replace with shorter github link after merge of https://github.com/evolution-gaming/skafka/pull/122
+  * {{{
+  *
+  * class SaveOffsetsOnRebalance[F[_]: Applicative] extends RebalanceListener1[F] {
+  *  import RebalanceCallback._
+  *
+  *  def onPartitionsAssigned(partitions: Nes[TopicPartition]) =
+  *    for {
+  *      // read the offsets from an external store using some custom code not described here
+  *      offsets <- lift(readOffsetsFromExternalStore[F](partitions))
+  *      _       <- offsets.toList.traverse_ { case (partition, offset) => seek(partition, offset) }
+  *    } yield ()
+  *
+  * def onPartitionsRevoked(partitions: Nes[TopicPartition]) =
+  *    for {
+  *      positions <- partitions.foldM(Map.empty[TopicPartition, Offset]) {
+  *        case (offsets, partition) =>
+  *          for {
+  *            position <- position(partition)
+  *          } yield offsets + (partition -> position)
+  *      }
+  *      // save the offsets in an external store using some custom code not described here
+  *      _ <- lift(saveOffsetsInExternalStore[F](positions))
+  *    } yield ()
+  *
+  *  // do not need to save the offsets since these partitions are probably owned by other consumers already
+  *  def onPartitionsLost(partitions: Nes[TopicPartition]) = empty
+  * }
+  *
+  * }}}
+  * @see [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]]
+  * @see [[com.evolutiongaming.skafka.consumer.RebalanceCallback]]
+  */
+trait RebalanceListener1[F[_]] { self =>
+
+  def onPartitionsAssigned(partitions: NonEmptySet[TopicPartition]): RebalanceCallback[F, Unit]
```

Comment:
should we use `Nes` as in other places for sake of consistency?

Reply:
this is a public api, I think we should use `NonEmptySet` instead of `Nes` there

Reply:
I'm fine on any of those options, I'm rather saying here about consistency across the project

Reply:
actually I changed it only for the sake of readability and documentation, 
so I don't have a strong opinion here about usage in the code,
but in docs we better use full name `NonEmptySet` (and I failed there, will fix shortly)

Reply:
fixed in ef133918